sklearn/metrics/_dist_metrics.pxd.tp

```diff
@@ -5,30 +5,26 @@ implementation_specific_values = [
     #
     #       name_suffix, INPUT_DTYPE_t, INPUT_DTYPE
     #
+    #
     # On the first hand, an empty string is used for `name_suffix`
-    # for the float64 case as to still be able to expose the original
-    # float64 implementation under the same API, namely `DistanceMetric`.
+    # for the 64bit case as to still be able to expose the original
+    # 64bit implementation under the same API, namely `DistanceMetric`.
     #
-    # On the other hand, '32' bit is used for `name_suffix` for the float32
+    # On the other hand, '32' is used for `name_suffix` for the float32
     # case to remove ambiguity and use `DistanceMetric32`, which is not
     # publicly exposed.
     #
     # The metric mapping is adapted accordingly to route to the correct
     # implementations.
     #
-    # We also use 64bit types as defined in `sklearn.utils._typedefs`
-    # to maintain backward compatibility at the symbol level for extra
-    # safety.
-    #
-    ('', 'DTYPE_t', 'DTYPE'),
-    ('32', 'cnp.float32_t', 'np.float32')
+    ('', 'float64_t', 'np.float64'),
+    ('32', 'float32_t', 'np.float32')
 ]
 
 }}
-cimport numpy as cnp
 from libc.math cimport sqrt, exp
 
-from ..utils._typedefs cimport DTYPE_t, ITYPE_t, SPARSE_INDEX_TYPE_t
+from ..utils._typedefs cimport float64_t, float32_t, int32_t, intp_t
 
 {{for name_suffix, INPUT_DTYPE_t, INPUT_DTYPE in implementation_specific_values}}
```

Comment:
Do we want to adapt the names of `INPUT_DTYPE_t` and `INPUT_DTYPE`?

Reply:
Do you have an idea ?
I thought about ``input_stype_t`` but it doesn't make a huge difference :)

Reply:
I think those are already good names.